rpc/test/test.cpp

```diff
@@ -485,6 +485,123 @@ TEST_F(RpcTest, passive_shutdown) {
     GTEST_ASSERT_LT(duration, 3500);
 }
 
+class RpcServerTimeout {
+public:
+    RpcServerTimeout(Skeleton* skeleton, net::ISocketServer* socket) : m_socket(socket), m_skeleton(skeleton) {
+        m_skeleton->register_service<Operation, OperationT>(this);
+        m_socket->set_handler({this, &RpcServerTimeout::serve});
+    }
+    struct Operation {
+        const static uint32_t IID = 0x1;
+        const static uint32_t FID = 0x2;
+        struct Request : public photon::rpc::Message {
+            int code = 0;
+            PROCESS_FIELDS(code);
+        };
+        struct Response : public photon::rpc::Message {
+            int code = 0;
+            PROCESS_FIELDS(code);
+        };
+    };
+
+    struct OperationT {
+        const static uint32_t IID = 0x1;
+        const static uint32_t FID = 0x3;
+        struct Request : public photon::rpc::Message {
+            int code = 0;
+            PROCESS_FIELDS(code);
+        };
+        struct Response : public photon::rpc::Message {
+            int code = 0;
+            PROCESS_FIELDS(code);
+        };
+    };
+    int do_rpc_service(Operation::Request* req, Operation::Response* resp, IOVector* iov, IStream* stream) {
+        resp->code = req->code;
+        return 0;
+    }
+    int do_rpc_service(OperationT::Request* req, OperationT::Response* resp, IOVector* iov, IStream* stream) {
+        resp->code = req->code;
+        photon::thread_usleep(5UL*1000*1000);
+        return 0;
+    }
+
+    int serve(photon::net::ISocketStream* stream) {
+        return m_skeleton->serve(stream);
+    }
+    int run() {
+        if (m_socket->bind_v4localhost() != 0)
+        // if (m_socket->bind(9527, net::IPAddr::V6Any()) != 0)
```

Comment:
remove unused code?

Reply:
Good idea

All unused code in this cpp has been removed.